lib/expressions/Term.ts

```diff
@@ -65,7 +68,9 @@ export class BlankNode extends Term {
 // Literals-- -----------------------------------------------------------------
 export class Literal<T> extends Term {
   public termType: 'literal' = 'literal';
-  public type: C.Type;
+  // TODO: Ask for PR: do we still need this type? I keep it just for NonLexicalLiteral right now.
+  //  Do we need the TypeAlias.SPARQL_NON_LEXICAL? We might be able to just use typeURL.value?
+  public type: LiteralTypes;
```

Comment:
Awesome, removing that would provided some even cleaner code. Personally I would go for `dataType` because I think `type` feels kind of empty. (It doesn't mean much)

Reply:
Agreed.

Reply:
I removed both `type` and `typeUrl` and replaced them by `dataType: string` and in certain Literals this becomes `dataType: LiteralTypes`. Since `LiteralTypes` is a subset of string I think this is a pretty solution.